### PR TITLE
Add documentation for log forwarding

### PR DESCRIPTION
**Pull request by sudomateo**
This adds documentation for log forwarding which is to be released in Terraform Enterprise v202109-1.

### CODE REVIEW

**Comment by sudomateo** [comment]
Fixing the link issue in #1852.

**Review by anniehedgpeth — CHANGES_REQUESTED** [review]
I was reviewing for readability and understandability, and it looked good! I have this one tiny picky change request. :)

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [inline content/source/docs/enterprise/admin/logging.html.md:8]: Should we include a version qualifier so users know it isn't available prior to v202109-1?

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [inline content/source/docs/enterprise/admin/logging.html.md:42]:
```suggestion
Enable log forwarding for an Active/Active installation of Terraform Enterprise:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [inline content/source/docs/enterprise/admin/logging.html.md:36]:
```suggestion
Enable log forwarding for a Standalone installation of Terraform Enterprise:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [inline content/source/docs/enterprise/admin/logging.html.md:75]:
```suggestion
Configure log forwarding for a Standalone installation of Terraform Enterprise:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [inline content/source/docs/enterprise/admin/logging.html.md:81]:
```suggestion
Configure log forwarding for an Active/Active installation of Terraform
```

**Review by aaron-lane — CHANGES_REQUESTED** [review]
Nice work! Just a few minor suggestions.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (aaron-lane) [reply content/source/docs/enterprise/admin/logging.html.md:42]: I think we try to capitalize the names of the operational modes 🤔

File: content/source/docs/enterprise/admin/logging.html.md
Comment (amy-hashi) [reply content/source/docs/enterprise/admin/logging.html.md:8]: We don't typically call that out. The assumption has been what's on the website is accurate for the current release.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (amy-hashi) [inline content/source/docs/enterprise/admin/logging.html.md:150]: We've got a lot of ~Notes~ going on, perhaps we move this (and the gcp one) just to the top of the appropriate section instead?

File: content/source/docs/enterprise/admin/logging.html.md
Comment (amy-hashi) [inline content/source/docs/enterprise/admin/logging.html.md:22]: Should we mention the default port here?

**Review by amy-hashi — CHANGES_REQUESTED** [review]

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Done!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: I'm a stickler for grammar things, so I'm happy to have this. Done!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: I moved this information to the top of its respective section and removed the note formatting for it. Let me know if that reads better.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md:22]: No need for this. All the connections are egress from Terraform Enterprise to the desired destination. The ports for the destinations can vary.

**Comment by sudomateo** [comment]
Noting that I'd like someone from Support to review as well before merging. I pinged them asking for someone to review.

**Review by duplo83 — APPROVED** [review]
Tons of great information here! You've preemptively answered many small questions, like install requirements and TFE restart necessity for logging to work.

**Review by aaron-lane — APPROVED** [review]

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:10]: I think we can omit the last sentence just because I think folks can get that from the page title. But! What I think we could write instead is maybe a sentence or two about why folks may want to do this? Something like "This is useful when you....... [x, y, z]"


```suggestion
Terraform Enterprise supports forwarding its logs to one or more external
destinations, a process called log forwarding. 
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:16]: If it's not enabled, should we tell them how to do that, or will they already know?

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:23]:
```suggestion
destination(s) where logs should be forwarded.
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:22]:
```suggestion
- Network connectivity between Terraform Enterprise and the external
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:30]: I wonder if we can just delete this Configuration heading and move the others up to be at level 2. I say this because this is the last level 2 on the page, so there's no need to separate out this content from...other level 2 content if that makes sense. I think it would be faster and clearer to just remove "configuration" and just have the descriptive headers you have below.

This note about needing to restart is important and I also wonder if we need to make it more salient/move it somewhere else where the user is more likely to see it/not miss it. Thoughts?

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:34]:
```suggestion
Log forwarding is disabled by default. To enable log forwarding, set the `log_forwarding_enabled` application setting to the value `1`.
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:36]:
```suggestion
For a Standalone installation of Terraform Enterprise:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:49]:
```suggestion
When log forwarding is enabled, the Terraform Enterprise application settings show the
following for `log_forwarding_enabled`:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:59]: Moved this above!
```suggestion
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:73]:
```suggestion
The `log_forwarding_config` Terraform Enterprise application setting must contain valid [Fluent Bit `[OUTPUT]` configuration](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit) specifying external destination(s) where Terraform Enterprise should forward logs. Since the Terraform Enterprise application settings are stored as JSON strings, we recommend first creating a `fluent-bit.conf` file with the valid Fluent Bit `[OUTPUT]` configuration and then using that file to configure the `log_forwarding_config` application setting. This method ensures that the configuration is stored in the application settings exactly how it appears in the `fluent-bit.conf` file.
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:89]:
```suggestion
Once configured, the Terraform Enterprise application settings show the
`log_forwarding_config` setting in escaped JSON string format:
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:125]:
```suggestion
-> **NOTE:** Do not use an `[OUTPUT]` directive with the
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:127]:
```suggestion
Doing this creates a loop that continuously emits logs!
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:132]: I wonder if we need to have a note or callout or something in the section about how to set up external forwarding (maybe in the first paragraph) to be like 'You have to pick one of the ones below..." or somehow call this out?

```suggestion
You can only forward logs to one of the supported external destinations below. Each
supported external destination contains example configuration for convenience.
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:153]:
```suggestion
-> **Note** In Terraform Enterprise installations using AWS external services,
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:177]:
```suggestion
-> **Note:** In Terraform Enterprise installations using AWS external services,
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:267]:
```suggestion
-> **Note:** In Terraform Enterprise installations using GCP external services,
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:314]:
```suggestion
If you have a requirement to split audit logs from application logs, we recommend forwarding all Terraform Enterprise logs to a log aggregation system, filtering the audit logs based on the `[Audit Log]` string, and forwarding just the audit logs to the desired destination.
```

File: content/source/docs/enterprise/install/automating-the-installer.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/install/automating-the-installer.html.md:162]:
```suggestion
- `log_forwarding_config` - (Optional) Valid [log forwarding](../admin/logging.html) configuration specifying external destination(s) to forward logs.
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Hi! Really great job on this! I love how you included so many examples. As promised, I took a pass through for style, flow, grammar, etc. See what you think! Once you look through everything I'd like to take one final pass :-)

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Love this! I removed the Configuration heading, moved the existing headings up to level 2, and added a Restart Terraform Enterprise heading.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: My latest edits do not include the "why?". I'll write that up and push changes. Leaving this open until then.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md:15]: I'm going to chew on this a bit. I don't want to overload that list item. Technically if a user is running on a support operating system then they should already be using `systemd-journald`. Leaving this open while I work on it.

Comment (laurapacilio) [reply content/source/docs/enterprise/admin/logging.html.md:15]: gotcha! Just let me know when it's ready for me to look again :-)

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:124]:
```suggestion
Once log forwarding is enabled and configured, you need to restart Terraform Enterprise for the changes to take effect. Please refer to the
[HashiCorp Help Center](https://support.hashicorp.com/hc/en-us/articles/360047602093-Restarting-Terraform-Enterprise)
for details on how to restart Terraform Enterprise.
```

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [reply content/source/docs/enterprise/admin/logging.html.md]: I like it!!! I did one tiny edit for you for style, but otherwise this is good, you can resolve!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md:15]: I changed up the wording a tiny bit and decide to leave out instructions on starting and enabling the `systemd-journald` service. That service should be started and enabled by default on our supported operating systems and if it is not then that's an outlier that should probably be funneled through the proper Customer Success channels.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Pushed up changes that cover the "why?". Let me know what you think.

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Accepted the edit! Thank you!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [reply content/source/docs/enterprise/admin/logging.html.md]: gotcha okay! works for me!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (laurapacilio) [inline content/source/docs/enterprise/admin/logging.html.md:11]:
```suggestion
destinations, a process called log forwarding. Log forwarding provides increased observability, assistance complying with log retention requirements, and information during troubleshooting.
```

**Review by laurapacilio — APPROVED** [review]
Really nice work on this! I left you a few more tiny nit comments, but I'm approving so I don't block. :-)

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md:56]: Done!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Done!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md]: Submitted the suggestion!

File: content/source/docs/enterprise/admin/logging.html.md
Comment (mikegreen) [inline content/source/docs/enterprise/admin/logging.html.md:20]: I feel the "which version, though?" question could be mitigated if we can add the specific version it started in. Is that a clear thing to add?

Comment (sudomateo) [reply content/source/docs/enterprise/admin/logging.html.md:20]: Based on a quick parsing of [Docker's changelog](https://github.com/moby/moby/blob/master/CHANGELOG.md), I see the earliest version of Docker to mention the `journald` logging driver is 1.9.0. However, Terraform Enterprise requires a minimum Docker version of 17.06 for the community edition and 1.13 for the version shipped in RHEL extras. That's why I figured running a command to verify the plugin is listed would be better than giving a version that may or may not work.

**Pull request merged**
